tools/link.py

```diff
@@ -172,7 +172,10 @@ def setup_environment_settings():
   if any(x for x in settings.ENVIRONMENT if x not in VALID_ENVIRONMENTS):
     exit_with_error(f'Invalid environment specified in "ENVIRONMENT": {settings.ENVIRONMENT}. Should be one of: {",".join(VALID_ENVIRONMENTS)}')
 
-  settings.ENVIRONMENT_MAY_BE_WEB = not settings.ENVIRONMENT or 'web' in settings.ENVIRONMENT or 'worker' in settings.ENVIRONMENT
+  if settings.ENVIRONMENT and settings.ENVIRONMENT == ['worker']:
```

Comment:
I guess we would want something more like 

`"worker" in settings.ENVIRONMENT and not any([e in settings.ENVIRONMENT for e in ['web', 'node'])`

And then the error message should explicitly say something like "specifying `worker` requires wither `node` or `web`"